src/main/java/io/vertx/core/http/impl/ClientHandler.java

```diff
@@ -107,14 +107,12 @@ protected void handleMessage(ClientConnection conn, ContextImpl context, Channel
         case BINARY:
         case CONTINUATION:
         case TEXT:
+        case PONG:
           conn.handleWsFrame(frame);
           break;
         case PING:
           // Echo back the content of the PING frame as PONG frame as specified in RFC 6455 Section 5.5.2
-          chctx.writeAndFlush(new WebSocketFrameImpl(FrameType.PONG, frame.getBinaryData()));
-          break;
-        case PONG:
-          // Just ignore it
+          chctx.channel().writeAndFlush(new PongWebSocketFrame(frame.getBinaryData().copy()));
```

Comment:
use `chctx` for not `ctx.channe()` it is more efficient.

Reply:
See comment on HttpServerImpl

Reply:
please really use chctx : see http://normanmaurer.me/presentations/2014-facebook-eng-netty/slides.html#29.0

Reply:
Yes, I changed this, per comment in ClientHandler; using `PongWebSocketFrame` is the real fix, not calling `channel()`